### PR TITLE
Ryna-LA100 GTAP downscale ctry

### PR DESCRIPTION
**Pull request by rynacui**

### CODE REVIEW

File: R/zchunk_LA100.GTAP_downscale_ctry.R [hunk 1]
```diff
@@ -32,43 +32,66 @@ module_aglu_LA100.GTAP_downscale_ctry_DISABLED <- function(command, ...) {
...
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Prepare land value for entire GTAP region
+    FAO_ag_items_PRODSTAT %>%
+      select(GTAP_use) %>%
+      unique() %>%
```
Comment (bpbond) [inline R/zchunk_LA100.GTAP_downscale_ctry.R]: Does `distinct` work here? If so please use instead.

File: R/zchunk_LA100.GTAP_downscale_ctry.R [hunk 2]
```diff
@@ -32,43 +32,66 @@ module_aglu_LA100.GTAP_downscale_ctry_DISABLED <- function(command, ...) {
...
+    L100.LDS_ag_prod_t %>%
+      left_join(FAO_ag_items_PRODSTAT[c("GTAP_crop", "GTAP_use")], by = "GTAP_crop") %>%
+      filter(!is.na(GTAP_use)) %>%
+      group_by(iso, GLU, GTAP_use) %>%
+      summarise_if(is.numeric, sum) %>%
+      ungroup() %>%
+      rename(prod_ctry = value) %>%
+      # Compute the country-within-GTAP region shares for each of the commodity classes
+      # Share = production by country and GTAP use / production by GTAP region and GTAP use
+      left_join_error_no_match(GTAP_ctry, by = "iso") %>%
```
Comment (bpbond) [inline R/zchunk_LA100.GTAP_downscale_ctry.R]: Great you're using these restrictive joins--thanks.

File: R/zchunk_LA100.GTAP_downscale_ctry.R [hunk 3]
```diff
@@ -32,43 +32,66 @@ module_aglu_LA100.GTAP_downscale_ctry_DISABLED <- function(command, ...) {
...
+      # Multiply the land values by the shares
+      right_join(LV_Rgtap, by = c("iso", "GTAP_region", "GLU", "GTAP_use")) %>%
+      mutate(value = value * share) %>%
+      mutate(value = if_else(is.na(value), 0, value)) %>%
+      select(-prod_ctry, -prod_rgn, -share) %>%
+
+      # Produce outputs
+      add_title("Land value by country / GLU / GTAP commodity class") %>%
+      add_units("Million US Dollars") %>%
+      add_comments("Compute the country-within-GTAP region's prodcution share for each of the commodity classes") %>%
```
Comment (bpbond) [inline R/zchunk_LA100.GTAP_downscale_ctry.R]: "production"

File: R/zchunk_LA100.GTAP_downscale_ctry.R [hunk 4]
```diff
@@ -12,9 +12,9 @@
 #' @importFrom assertthat assert_that
```
Comment (bpbond) [inline R/zchunk_LA100.GTAP_downscale_ctry.R]: Do you fill in a `@details` field? If not, I think probably would be good to do.

File: R/zchunk_LA100.GTAP_downscale_ctry.R [hunk 5]
```diff
@@ -32,43 +33,66 @@ module_aglu_LA100.GTAP_downscale_ctry_DISABLED <- function(command, ...) {
...
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Prepare land value for entire GTAP region
+    FAO_ag_items_PRODSTAT %>%
+      select(GTAP_use) %>%
+      distinct() %>%
+      na.omit() %>%
+      repeat_add_columns(distinct(L100.LDS_ag_prod_t[c("iso", "GLU")])) %>%
+      left_join_error_no_match(GTAP_ctry, by = "iso") %>%
+      # Match in the land value for the entire GTAP region. These will be multiplied by country shares
```
Comment (kvcalvin) [inline R/zchunk_LA100.GTAP_downscale_ctry.R]: What are GTAP regions?

**Comment by bpbond** [comment]
@kvcalvin OK to merge?

**Comment by bpbond** [comment]
Closes #116

**Pull request merged**